src/main/java/pl/edukacyjni/user/UserService.java

```diff
@@ -5,19 +5,41 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class UserService {
 
     private static final List<User> USERS = new ArrayList<>(){{
-        add(new User(1L, "michaelPaello", "kochamPierogi"));
-        add(new User(2L, "theGhostCasper", "huliganJezusaSt0"));
-        add(new User(3L, "stNicolas", "haslo123!"));
-        add(new User(4L, "santiago", "deadline"));
-        add(new User(5L, "was-far", "abcd"));
+        add(new User(1L, "michaelPaello", "kochamPierogi", false));
+        add(new User(2L, "theGhostCasper", "huliganJezusaSt0", true));
+        add(new User(3L, "stNicolas", "haslo123!", false));
+        add(new User(4L, "santiago", "deadline", true));
+        add(new User(5L, "was-far", "abcd", true));
     }};
 
     public List<User> getAllUsers() {
         return USERS;
     }
+
+    public Optional<User> getUserById(Long id) {
+        for (User user : USERS) {
+            if (id.equals(user.getId())) {
+                return Optional.of(user);
+            }
+        }
+        return Optional.empty();
+    }
+
+    public Optional<User> deactivateUserById(Long id) {
```

Comment:
I like that you added that option but IMHO we should separate it. This should be a command so the result should be void. Please check _command query separation_.